responses.py

```diff
@@ -402,9 +408,42 @@ def get_response(self, request):
             body=body,
             headers=headers,
             preload_content=False,
+            original_response=_OriginalResponse(headers),
         )
 
 
+class _OriginalResponse(object):
+    def __init__(self, headers):
+        headers = headers.copy()
+
+        if "set-cookie" in headers:
+            cookiejar = _cookies_from_headers(headers)
+            headers.pop("set-cookie", None)
+        else:
+            cookiejar = None
+
+        if six.PY2:
+            items = [k + ": " + v for k, v in headers.iteritems()]
```

Comment:
This looks like it could hit a UnicodeError if the header contains a non-ascii byte.

Reply:
Could you first take a look at #269, which I just created? I was adding a couple more tests on headers/cookie handling and that would help me better understand how to go about handling this issue.